File: backend/activity/src/main/java/leavelive/activity/config/SecurityConfig.java
package leavelive.activity.config;

import leavelive.activity.config.jwt.JwtAuthorizationFilter;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.config.http.SessionCreationPolicy;
import org.springframework.web.filter.CorsFilter;

@Configuration
@EnableWebSecurity
@RequiredArgsConstructor
public class SecurityConfig extends WebSecurityConfigurerAdapter {
    private final CorsFilter corsFilter;

    @Override
    protected void configure(HttpSecurity http) throws Exception {
        http.csrf().disable();
        http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS) //세션 사용 X
                .and()
                .addFilter(corsFilter)
                .formLogin().disable()
                .addFilter(new JwtAuthorizationFilter(authenticationManager()))
                .httpBasic().disable()
                .authorizeRequests()
                .anyRequest().permitAll();
    }
}

File: backend/kakao_login/authentication/src/main/kotlin/com/ssafy/authentication/config/SecurityConfig.kt
package com.ssafy.authentication.config

import com.ssafy.authentication.filter.JwtFilter
import com.ssafy.authentication.util.JwtTokenManager
import org.springframework.context.annotation.Bean
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.builders.WebSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
import org.springframework.security.config.http.SessionCreationPolicy
import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter
import org.springframework.web.cors.CorsConfiguration
import org.springframework.web.cors.CorsConfigurationSource
import org.springframework.web.cors.UrlBasedCorsConfigurationSource
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@EnableWebSecurity
class SecurityConfig(private val jwtTokenManager: JwtTokenManager) : WebSecurityConfigurerAdapter() {

    override fun configure(http: HttpSecurity) {
        //super.configure(http)
        http.httpBasic().disable().csrf().disable().cors()
        http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
            .and()
            .authorizeRequests()
            .antMatchers("/api/**").permitAll()
            .and()
            .addFilterBefore(JwtFilter(jwtTokenManager), UsernamePasswordAuthenticationFilter::class.java)

    }

    override fun configure(web: WebSecurity) {
        web.ignoring().antMatchers("/auth/**")
    }

    @Bean
    fun corsConfigurationSource(): CorsConfigurationSource {
        val config = CorsConfiguration()
        config.allowedOrigins = listOf("*")
        config.addAllowedHeader("*")
        config.addAllowedMethod("*")
        //config.allowCredentials = true

        val src = UrlBasedCorsConfigurationSource()
        src.registerCorsConfiguration("/**", config)
        return src
    }

    @Bean
    override fun authenticationManager(): AuthenticationManager {
        return super.authenticationManager()
    }
}